web_external/models/UserModel.js

```diff
@@ -33,39 +34,53 @@ UserModel.prototype.canAcceptTerms = function () {
     return this.get('permissions').acceptTerms === true;
 };
 
-UserModel.prototype.setAcceptTerms = function (successCallback) {
+UserModel.prototype.setAcceptTerms = function () {
+    const deferred = $.Deferred();
     restRequest({
         path: 'user/acceptTerms',
         type: 'POST'
-    }).done((resp) => {
+    })
+    .done((resp) => {
         if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
             // Directly update user permissions
             this.get('permissions').acceptTerms = true;
             this.trigger('change:permissions');
-            successCallback(resp);
+            deferred.resolve(resp);
+        } else {
+            // This should not fail
+            deferred.reject(resp);
         }
-        // This should not fail
+    })
+    .fail((resp) => {
+        deferred.reject(resp);
     });
+    return deferred.promise();
 };
 
 UserModel.prototype.canCreateDataset = function () {
     return this.get('permissions').createDataset;
 };
 
-UserModel.prototype.setCanCreateDataset = function (successCallback, failureCallback) {
+UserModel.prototype.setCanCreateDataset = function () {
+    const deferred = $.Deferred();
     restRequest({
         path: 'user/requestCreateDatasetPermission',
         type: 'POST'
-    }).done((resp) => {
+    })
+    .done((resp) => {
         if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
             // Directly update user permissions
             this.get('permissions').createDataset = true;
             this.trigger('change:permissions');
-            successCallback(resp);
+            deferred.resolve(resp);
         } else {
-            failureCallback(resp);
+            deferred.reject(resp);
         }
+    })
+    .fail((resp) => {
+        deferred.reject(resp);
     });
+    return deferred.promise();
```

Comment:
Totally agreed. jQuery 2 is a little weird in how it handles `.then`, so I wanted to be safe, but just returning a thenable should still work.